File: Python/2019/S1.py
# PASSED

translations = list(input())
V, H = translations.count('V'), translations.count('H')

grid = [
    [1, 2],
    [3, 4]
]

for _ in range(V % 2):
    grid[0][0], grid[0][1] = grid[0][1], grid[0][0]
    grid[1][0], grid[1][1] = grid[1][1], grid[1][0]


for _ in range(H % 2):
    grid[0][0], grid[1][0] = grid[1][0], grid[0][0]
    grid[0][1], grid[1][1] = grid[1][1], grid[0][1]

for q in grid:
    print(*q)

File: C++/2019/S1.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int a = 1, b = 2, c = 3, d = 4, temp;
    string input;cin >> input;

    int h = count(input.begin(), input.end(), 'H') % 2;
    int v = count(input.begin(), input.end(), 'V') % 2;

    if (h) {
        temp = a;
        a = c;
        c = temp;

        temp = b;
        b = d;
        d = temp;
    }
    if (v) {
        temp = a;
        a = b;
        b = temp;

        temp = c;
        c = d;
        d = temp;
    }

    cout << a << ' ' << b << endl << c << ' ' << d;
}